src/main/java/com/api/Comparator.java

```diff
@@ -8,4 +10,10 @@
         return (left, right) -> compare(right, left);
     }
 
+    default Comparator<T> thenComparing(Comparator<? super T> other) {
+        requireNonNull(other);
+        return (left, right) -> (
+                (compare(left, right) != 0) ? compare(left, right) : other.compare(left, right)
```

Comment:
problem li kayen hnaya, c'est que tu as appelé `compare(left, right)` dial `this` deux fois :sssss

Reply:
daba ila kan compare(left, right) dial this stateful, rah il peut retourner une valeur differrente de 0 la premiere fois, mais la deuxième fois une valeur = 0

Reply:
matenssach compare(left, right) rah elle va invoquer une méthode, donc sera mieux de cacher le résultat pour utiliser le même